lib/pages/settings.dart

```diff
@@ -126,6 +153,137 @@ class _AccountsConfig extends StatelessWidget {
+                        .addInstance(instanceController.text);
+                    scaffoldKey.currentState.hideCurrentSnackBar();
+                  } on Exception catch (err) {
+                    scaffoldKey.currentState.showSnackBar(SnackBar(
+                      content: Text(err.toString()),
+                    ));
+                  }
+                  loading.value = false;
+                  Navigator.of(context).pop();
+                },
+        ),
+        FlatButton(
+          child: Text('Cancel'),
+          onPressed: () {
+            Navigator.of(context).pop();
+          },
+        )
+      ],
+    );
+  }
+}
+
+class _AccountsConfigAddAccountDialog extends HookWidget {
+  final GlobalKey<ScaffoldState> scaffoldKey;
+  final String instanceUrl;
+
+  const _AccountsConfigAddAccountDialog(
+      {@required this.scaffoldKey, @required this.instanceUrl})
+      : assert(scaffoldKey != null),
+        assert(instanceUrl != null);
+
+  @override
+  Widget build(BuildContext context) {
+    var usernameController = useTextEditingController();
+    var passwordController = useTextEditingController();
+    useValueListenable(usernameController);
+    useValueListenable(passwordController);
+
+    var loading = useState(false);
+
+    return AlertDialog(
+      title: Text('Add account'),
+      content: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          TextField(
+            autofocus: true,
+            controller: usernameController,
+            decoration: InputDecoration(
+              border: OutlineInputBorder(),
+              labelText: 'Username or email',
+            ),
+          ),
+          const SizedBox(height: 5),
+          TextField(
+            controller: passwordController,
+            obscureText: true,
+            decoration: InputDecoration(
+              border: OutlineInputBorder(),
+              labelText: 'Password',
+            ),
+          ),
+        ],
+      ),
+      actions: <Widget>[
+        FlatButton(
+          child: !loading.value ? Text('Add') : CircularProgressIndicator(),
+          onPressed:
+              usernameController.text.isEmpty || passwordController.text.isEmpty
+                  ? null
+                  : () async {
+                      try {
+                        loading.value = true;
+                        await context.read<AccountsStore>().addAccount(
+                              instanceUrl,
+                              usernameController.text,
+                              passwordController.text,
+                            );
+                      } on Exception catch (err) {
+                        scaffoldKey.currentState.showSnackBar(SnackBar(
+                          content: Text(err.toString()),
+                        ));
+                      }
+                      loading.value = false;
+                      Navigator.of(context).pop();
+                    },
```

Comment:
should be separated to a function. not be directly in UI. can be in the build method, just not in the UI